temporal-sdk/src/main/java/io/temporal/worker/Worker.java

```diff
@@ -561,15 +561,23 @@ private static SingleWorkerOptions toLocalActivityOptions(
         .build();
   }
 
+  @SuppressWarnings("deprecation")
   private static SingleWorkerOptions.Builder toSingleWorkerOptions(
       WorkerFactoryOptions factoryOptions,
       WorkerOptions options,
       WorkflowClientOptions clientOptions,
       List<ContextPropagator> contextPropagators) {
+    String buildID = null;
+    if (options.getBuildID() != null) {
+      buildID = options.getBuildID();
+    } else if (clientOptions.getBinaryChecksum() != null) {
+      buildID = clientOptions.getBinaryChecksum();
+    }
```

Comment:
```suggestion
    String buildID = clientOptions.getBinaryChecksum();
    if (options.getBuildID() != null) {
      buildID = options.getBuildID();
    }
```

(style choice, doesn't really matter)